.github/workflows/DetectCDKBootstrapVersionChanges.yml

```diff
@@ -1,49 +1,107 @@
 name: Detect CDK Bootstrap Version Changes
 
-on: [pull_request]
+on:
+  schedule:
+    # Runs at 00:00 UTC every Monday
+    - cron: '0 0 * * 1'
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+  id-token: write
 
 jobs:
   detect-cdk-bootstrap-changes:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@ececac1a45f3b08a01d2dd070d28d111c5fe6722
```

Comment:
nit: next to the commit id we typically add a comment to refer to which version this is